### PR TITLE
Enable 1ES pools for live tests.

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR adds support for 1ES pools on live test pipelines. It does this by adding a ```Pool``` parameter to platform entries which is then propagated down into the test job template as a variable in the matrix (which is then used in the ```pool:``` setting on jobs.

### CODE REVIEW

**Comment by mitchdenny** [comment]
@benbp this largely works but the issue that I'm running into is that the identity test pipeline uses a container image which doesn't have .NET 5.0 on it. The host agents do now have .NET 5.0 and our install steps have been commented out. I think the right course of action here is to rev the container image that we are using.

**Comment by benbp** [comment]
@mitchdenny the container issue to update to net50 is being worked on by @schaabs

**Comment by benbp** [comment]
https://github.com/Azure/azure-sdk-for-net/pull/17980 @mitchdenny

File: eng/pipelines/templates/stages/archetype-sdk-tests.yml [hunk 1]
```diff
@@ -56,29 +56,36 @@ parameters:
   type: object
   default:
     Linux:
-      OSVmImage: "ubuntu-18.04"
+      Pool: azsdk-pool-mms-ubuntu-1804-general
+      OSVmImage: MMSUbuntu18.04
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-sdk-tests.yml:60]: Do the pools now support image names?

Comment (mitchdenny) [reply eng/pipelines/templates/stages/archetype-sdk-tests.yml:60]: I don't believe it does (yet).

**Pull request merged**